File: 백준/Bronze/1157. 단어 공부/단어 공부.py
words = input().upper()
#입력받은 문자열에서 중복값을 제거
unique_words = list(set(words))

cnt_list = []
for x in unique_words:
    #원래 문자열에서 각 알파벳들이 몇 개인지 count 해준다.
    cnt = words.count(x)
    #count 숫자를 리스트에 append
    cnt_list.append(cnt)

#알파벳 개수가 가장 큰 것이 여러개이면
if cnt_list.count(max(cnt_list)) > 1:
    print("?")

#그렇지 않으면, 가장 많이 사용된 알파벳을 대문자로 출력 
else:
    max_index = cnt_list.index(max(cnt_list))
    print(unique_words[max_index])

File: 백준/Bronze/1157. 단어 공부/단어 공부.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.HashMap;
import java.util.StringTokenizer;

public class Main {

    public static void main(String[] args) {
        FastScan fastScan = new FastScan();
        HashMap<Character, Integer> wordCountMap = new HashMap<>();

        int max = 0;
        int maxCnt = 0;

        String input = fastScan.next().toUpperCase();

        for (char c : input.toCharArray()) {
            wordCountMap.put(c, wordCountMap.getOrDefault(c, 0) + 1);
        }

        String ans = "";
        for (char c : wordCountMap.keySet()){
            if (wordCountMap.get(c) > max) {
                max = wordCountMap.get(c);
                ans = String.valueOf(c);
            }

        }

        for (int cnt : wordCountMap.values()) {
            if (max == cnt) maxCnt++;
        }

        ans = maxCnt > 1 ? "?" : ans;
        System.out.println(ans);
    }


    static class FastScan {
        BufferedReader br;
        StringTokenizer st;

        public FastScan() {
            this.br = new BufferedReader(new InputStreamReader(System.in));
        }

        String next() {
            if (st == null || !st.hasMoreElements()) {
                try {
                    st = new StringTokenizer(br.readLine());
                } catch (IOException e) {
                    e.printStackTrace();
                }
            }
            return st.nextToken();
        }
    }

}
